test/Interop/C/bounds-safety/Inputs/bounds-attributed-function.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <ptrcheck.h>
+#include <stdarg.h>
+
+#ifndef __unsafe_late_const
+#define __unsafe_late_const
+#endif
+
+int * __counted_by(len) a(int *__counted_by(len) p, int len);
+char * __counted_by(len) b(char *__counted_by(len) p, int len);
+char * __sized_by(len) c(char *__sized_by(len) p, int len);
+void * __sized_by(len) d(void *__sized_by(len) p, int len);
+int * __sized_by(len) e(int *__sized_by(len) p, int len);
+
+int * __ended_by(end) f(int *__ended_by(end) p, int * end);
+void * __ended_by(end) g(void *__ended_by(end) p, void * end);
+
+char * __null_terminated h(char *__null_terminated p);
+const char * i(const char * p);
+
+char * __single j(char *__single p);
+void *__single k(void *__single p);
+
+#if __has_ptrcheck
+char * __indexable l(char *__indexable p);
+void *__indexable m(void *__indexable p);
+
+char * __bidi_indexable n(char *__bidi_indexable p);
+void * __bidi_indexable o(void *__bidi_indexable p);
+#endif
+
+#if !__cplusplus
+void p(int len, int p[len]);
```

Comment:
Why this isn't allowed in C++? (And can you add a comment with your explanation?)

Reply:
VLAs are a C only feature